lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb

```diff
@@ -42,20 +42,33 @@ def self.integrate(pod_bundle, targets)
           #         The build configuration.
           #
           def self.set_target_xcconfig(pod_bundle, target, config)
-            path = pod_bundle.xcconfig_relative_path(config.name)
-            group = config.project['Pods'] || config.project.new_group('Pods')
-            file_ref = group.files.find { |f| f.path == path }
+            # Xcode root group's path is absolute, we must get the relative path of the sandbox to the user project
+            group_path = pod_bundle.relative_pods_root_path
+            group = config.project['Pods'] || config.project.new_group('Pods', group_path)
+
+            # support user custom paths of Pods group and xcconfigs files.
+            group_path = Pathname.new(group.path || '')
```

Comment:
this looks like a bunch of complicated logic, could you extract it out to its own method?

Reply:
done.